client/src/auth/sign-up.tsx

```diff
@@ -117,65 +121,92 @@ export default function SignUpApp() {
     }
   }
 
+  const userDetails: UserDetails = JSON.parse(
+    searchParams.get("user_details") || ""
```

Comment:
Yeah, that 
```suggestion
    searchParams.get("user_details") || "{}"
```
is actually the right thing to do but it's not entirely solving the problem because it only makes things OKish if the `user_details` is entirely missing.